### PR TITLE
MULE-19380: Tooling: Do a proper lazy init in AbstractArtifactAgnosticService

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/AbstractArtifactAgnosticService.java [hunk 1]
```diff
@@ -63,7 +72,26 @@ protected Application getStartedApplication() throws ApplicationStartingExceptio
...
         application.install();
-        application.init();
+
+        if (isLazyInit()) {
+          try {
+            application.lazyInit(!isLazyInitEnableXmlValidations());
+            // Force the initialization of the app components so that the registry is populated with the configs
+            application.getRegistry().lookupByType(LazyComponentInitializer.class)
+                .ifPresent(lazyInit -> lazyInit
+                    .initializeComponents((Predicate<ComponentAst>) comp -> comp.getModel(ConfigurationModel.class).isPresent()));
```
Comment (gsfernandes) [inline modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/AbstractArtifactAgnosticService.java:82]: We do need to enable also every global element besides Configurations.

Comment (gsfernandes) [reply modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/AbstractArtifactAgnosticService.java:82]: The idea at this point is that the application should only has globalElements (topLevel pojos) or configurations.

**Review by gsfernandes — CHANGES_REQUESTED** [review]

**Comment by elrodro83** [comment]
--validate

**Review by gsfernandes — APPROVED** [review]

**Comment by elrodro83** [comment]
--validate

**Review by gsfernandes — APPROVED** [review]

**Pull request merged**